source/slang/slang-ir-lower-generics.cpp

```diff
@@ -0,0 +1,159 @@
+// slang-ir-lower-generics.cpp
+#include "slang-ir-lower-generics.h"
+
+#include "slang-ir.h"
+#include "slang-ir-clone.h"
+#include "slang-ir-insts.h"
+
+namespace Slang
+{
+    struct GenericsLoweringContext;
+
+    struct GenericsLoweringContext
+    {
+        // For convenience, we will keep a pointer to the module
+        // we are specializing.
+        IRModule* module;
+
+        Dictionary<IRInst*, IRInst*> loweredGenericFunctions;
+
+        SharedIRBuilder sharedBuilderStorage;
+
+        // We will use a single work list of instructions that need
+        // to be considered for specialization or simplification,
+        // whether generic, existential, etc.
+        //
```

Comment:
Some of these comments seem to be copy-pasted from the specialization pass, and we might want to tailor them a bit more to what is going on.